lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -367,6 +384,115 @@ static LogicalResult verifyCircuitOp(CircuitOp circuit) {
       }
     }
   }
+  leafInnerSyms.sort();
+  instanceSyms.sort();
+
+  // Verify the NonLocalAnchor.
+  // 1. Iterate over the namepath.
+  // 2. The namepath should be a valid instance path, specified either on a
+  // module or a declaration inside a module.
+  // 3. Each element in the namepath is an InnerRefAttr except possibly the
+  // last element.
+  // 4. Make sure that the InnerRefAttr is legal, by verifying the module name
+  // and the corresponding inner_sym on the instance.
+  // 5. Make sure that the instance path is legal, by verifying the sequence of
+  // instance and the expected module occurs as the next element in the path.
+  // 6. The last element of the namepath, can be an InnerRefAttr on either a
+  // module port or a declaration inside the module.
+  // 7. The last element of the namepath can also be a module symbol.
+  auto hasNonLocal = [&](AnnotationSet &annos, StringRef nlaName) {
+    bool instFound = false;
+    for (auto anno : annos) {
+      if (auto nlaRef = anno.getMember("circt.nonlocal")) {
+        instFound = nlaRef.cast<FlatSymbolRefAttr>().getAttr() == nlaName;
+        if (instFound)
+          break;
+      }
+    }
+    return instFound;
+  };
+  for (NonLocalAnchor nla : nlaList) {
+    auto namepath = nla.namepath();
+    StringAttr nlaName = nla.sym_nameAttr();
+    if (namepath.size() <= 1) {
+      nla.emitOpError()
+          << "the instance path cannot be empty/single element, it "
+             "must specify an instance path.";
+      return failure();
```

Comment:
I think errors are implicitly convertible to `LogicalResult`, if this trick works, it can be used in a bunch of places.
```c++
      return nla.emitOpError()
          << "the instance path cannot be empty/single element, it "
             "must specify an instance path.";
```